### ISSUE TITLE
HandleHeaderLinks not working properly with QueryParams

### ISSUE THREAD

**Issue by AndreasBieber** [1and1/TypedRest-DotNet#9]
Hi Basti,

i've tried to implement the link templates. But the method `HandleHeaderLinks` cannot deal with templates containing query parameters.

Cheers
 Andy

**Comment by bastianeicher** [comment]
Hi Andy,

thanks for reporting this.

The HTTP spec allows for two different encodings of multi-value headers like `Link`:
 * The same header name can be listed multiple times, for example:
```
HTTP/1.1 200 OK
Link: <href1>; rel=rel1
Link: <href2>; rel=rel2
```
 * The values can be aggregated into a single header and separated with a comma, for example:
```
HTTP/1.1 200 OK
Link: <href1>; rel=rel1, <href2>; rel=rel2
```
Unfortunately, the .NET `HttpClient` does not take care of these details.

My code is currently missing proper escaping logic that ignores `;` and `,` between `<` and `>`. The `,` specifically is causing problems with URI Templates with multiple optional parameters.

I'll work on improving the logic in `HandleHeaderLinks()` to properly handle all these cases

Regards
Bastian

**Comment by AndreasBieber** [comment]
Hi Basti,

ah ok. Didn't knew this HTTP spec.

Thank you for the information.

Cheers
Andy

**Comment by bastianeicher** [comment]
Relevant specs for reference:
[HTTP Message Syntax and Routing, RFC 7320, Section 3.2.2](https://tools.ietf.org/html/rfc7230#section-3.2.2)
[Web Linking, RFC 5988, Section 5](https://tools.ietf.org/html/rfc5988#section-5)

This may take a little. :(

**Comment by AndreasBieber** [comment]
Do you want to implement the whole spec (regarding the fields)?

To just fix the problem, a simple regex could solve the problem:

```cs
var linkHeaderMatches = Regex.Matches("<http://localhost/endpoint{?name,email}>; rel=\"child\", <http://localhost/bla>; rel=\"foo\"", "(<[^>]+>;[^,]+)");
```

**Comment by bastianeicher** [comment]
Fixed by #10

Thanks for the pull request Andy :)
